sources/ecommerce-app/src/util/component.js

```diff
@@ -75,7 +79,7 @@ export function useUser({ redirect, onMissing } = {}) {
           onMissing && onMissing();
       }
     },
-    []
+    [dispatch, user, onMissing, redirect]
```

Comment:
Is this safe? Are consumers of useUser memoizing redirect & onMissing?

Reply:
using refs